Possible panic in glob.Glob.

The following input causes `glob.Glob` to panic with `runtime error: slice bounds out of range`:

```Go
package main

import "github.com/ryanuber/go-glob"

func main() {
	glob.Glob("*.gen.go", "")
}
```

```
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
panic(0x15a380, 0xc82000a090)
	/usr/local/go/src/runtime/panic.go:464 +0x3e6
github.com/ryanuber/go-glob.Glob(0x18d2b0, 0x8, 0x0, 0x0, 0xc820000300)
	src/github.com/ryanuber/go-glob/glob.go:61 +0x4cb
main.main()
	src/github.com/shurcooL/play/31/main.go:38 +0x38
exit status 2
```

It happens on [line 54 of glob.go](https://github.com/ryanuber/go-glob/blob/0067a9abd927e50aed5190662702f81231413ae0/glob.go#L54):

```Go
// Trim evaluated text from subj as we loop over the pattern.
idx := strings.Index(subj, part) + len(part) // At this time, subj is "", part is ".gen.go", so idx becomes 6.
subj = subj[idx:] // Since subj is "", doing subj[6:] is a slice bounds out of range panic.
```

I can work on a fix and create a PR to resolve this.

--- a/glob.go
+++ b/glob.go
@@ -40,9 +40,7 @@ func Glob(pattern, subj string) bool {
 				return false
 			}
 		case end:
-			if len(subj) > 0 {
-				return trailingGlob || strings.HasSuffix(subj, part)
-			}
+			return trailingGlob || strings.HasSuffix(subj, part)
 		default:
 			if !strings.Contains(subj, part) {
 				return false
@@ -54,6 +52,7 @@ func Glob(pattern, subj string) bool {
 		subj = subj[idx:]
 	}
 
-	// All parts of the pattern matched
-	return true
+	// We should never get here because "case end:" above should always trigger at the end of
+	// iterating over parts. If we do, it's a bug in this library that should be reported and fixed.
+	panic("unreachable")
 }

--- a/glob_test.go
+++ b/glob_test.go
@@ -1,6 +1,9 @@
 package glob
 
-import "testing"
+import (
+	"strings"
+	"testing"
+)
 
 func testGlobMatch(t *testing.T, pattern, subj string) {
 	if !Glob(pattern, subj) {
@@ -17,6 +20,19 @@ func testGlobNoMatch(t *testing.T, pattern, subj string) {
 func TestEmptyPattern(t *testing.T) {
 	testGlobMatch(t, "", "")
 	testGlobNoMatch(t, "", "test")
+}
+
+func TestEmptySubject(t *testing.T) {
+	testGlobMatch(t, "", "")
+	testGlobNoMatch(t, "test", "")
+	testGlobMatch(t, "*", "")
+	testGlobMatch(t, "**", "")
+	testGlobMatch(t, "***", "")
+	testGlobMatch(t, "******************************", "")
+	testGlobMatch(t, strings.Repeat("*", 1000000), "")
+	testGlobNoMatch(t, strings.Repeat("*", 1000000)+"x", "")
+	testGlobNoMatch(t, "x"+strings.Repeat("*", 1000000), "")
+	testGlobNoMatch(t, "x"+strings.Repeat("*", 1000000)+"x", "")
 }
 
 func TestPatternWithoutGlobs(t *testing.T) {

